src/app/containers/Rewards/utils/calculatePoolData.ts

```diff
@@ -0,0 +1,340 @@
+    sum.add(b)
+  );
+  const lpTokenPriceUSD = tokenBalancesSum.isZero()
+    ? Zero
+    : tokenBalancesUSDSum.mul(BigNumber.from(10).pow(18)).div(tokenBalancesSum);
+
+  function calculatePctOfTotalShare(lpTokenAmount: BigNumber): BigNumber {
+    const baseCalc =
+      useMasterchef && masterchefApr
+        ? BigNumber.from(masterchefApr[POOL.address].totalStaked)
+        : totalLpTokenBalance;
+
+    // returns the % of total lpTokens
+    return lpTokenAmount
+      .mul(BigNumber.from(10).pow(18))
+      .div(baseCalc.isZero() ? BigNumber.from("1") : baseCalc);
+  }
+
+  // lpToken balance in wallet as a % of total lpTokens, plus lpTokens staked elsewhere
+  const userShare = calculatePctOfTotalShare(userLpTokenBalance);
+  const userPoolTokenBalances = tokenBalances.map((balance) => {
+    return userShare.mul(balance).div(BigNumber.from(10).pow(18));
+  });
+  const userPoolTokenBalancesSum: BigNumber = userPoolTokenBalances.reduce(
+    (sum, b) => sum.add(b)
+  );
+  const userPoolTokenBalancesUSD = tokenBalancesUSD.map((balance) => {
+    return userShare.mul(balance).div(BigNumber.from(10).pow(18));
+  });
+  const userPoolTokenBalancesUSDSum: BigNumber =
+    userPoolTokenBalancesUSD.reduce((sum, b) => sum.add(b));
+
+  const poolTokens = effectivePoolTokens.map((token, i) => ({
+    symbol: token.symbol,
+    percent: formatBNToPercentString(
+      tokenBalances[i]
+        .mul(10 ** 5)
+        .div(
+          totalLpTokenBalance.isZero() ? BigNumber.from("1") : tokenBalancesSum
+        ),
+      5
+    ),
+    value: tokenBalances[i],
+  }));
+  const userPoolTokens = effectivePoolTokens.map((token, i) => ({
+    symbol: token.symbol,
+    percent: formatBNToPercentString(
+      tokenBalances[i]
+        .mul(10 ** 5)
+        .div(
+          totalLpTokenBalance.isZero() ? BigNumber.from("1") : tokenBalancesSum
+        ),
+      5
+    ),
+    value: userPoolTokenBalances[i],
+  }));
+  const poolAddress = POOL.address;
+  const { oneDayVolume, apr, utilization } =
+    swapStats && poolAddress in swapStats
+      ? swapStats[poolAddress]
+      : { oneDayVolume: 0, apr: 0, utilization: null };
+  const poolData = {
+    name: poolKey,
+    rapr: poolApr,
+    tokens: poolTokens,
+    reserve: tokenBalancesUSDSum,
+    totalLocked: totalLpTokenBalance,
+    virtualPrice: virtualPrice,
+    adminFee: adminFee,
+    swapFee: swapFee,
+    aParameter: aParameter,
+    volume: oneDayVolume,
+    utilization: utilization ? parseUnits(utilization, 18) : null,
+    apr: apr,
+    extraapr: extraapr,
+    lpTokenPriceUSD,
+    lpToken: POOL.lpToken.symbol,
+    isPaused,
+  };
+  const userShareData = account
+    ? {
+        name: poolKey,
+        share: userShare,
+        underlyingTokensAmount: userPoolTokenBalancesSum,
+        usdBalance: userPoolTokenBalancesUSDSum,
+        tokens: userPoolTokens,
+        lpTokenBalance: userLpTokenBalance,
+        masterchefBalance: userMasterchefBalances,
+      }
+    : null;
```

Comment:
`typescript` works better with `undefined` values than `null` we should consider coming up with a  consistent way of defining absent values.

Reply:
I prefer the undefined
this part is based on old Axial base code